### PR TITLE
Add knockbackOnHit

### PR DESCRIPTION
**Pull request by Koud-Wind**
## 🤔 What type of PR is this? (check all applicable)

- [x] 🍕 Addition
- [ ] ⌨️ Productivity
- [ ] 🐛 Bug Fix
- [ ] 🔥 Optimization
- [x] ⚙️ Configuration
- [ ] 🌟 Quality Of Life
- [ ] ✨ Enhancement
- [ ] 📝 Documentation

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by Desoroxxx** [comment]
Please clean this pr, there is way too many things not related to `knockbackOnHit` also the comment should be something like that:
`comment = "Whether entities should have knockback applied to them when shot?"`

**Comment by Koud-Wind** [comment]
Please hurry up

File: src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java [hunk 1]
```diff
@@ -136,13 +137,16 @@ protected void onImpact(RayTraceResult position) {
...
 
             if(this.getThrower() != null) {
-                position.entityHit.attackEntityFrom(DamageSource.causeThrownDamage(this, this.getThrower()), damage);
+				position.entityHit.attackEntityFrom(DamageSource.causeThrownDamage(this, this.getThrower()), damage);
             } else {
                 position.entityHit.attackEntityFrom(new DamageSource("thrown"), damage);
             }
 
+			if(!ModernConfigManager.knockbackOnHit)
+				position.entityHit.setVelocity(this.motionX / 1500, this.motionY / 1500, this.motionZ / 1500);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java:149]: Please fix the spacing

File: src/main/java/com/paneedah/weaponlib/config/ModernConfigManager.java [hunk 2]
```diff
@@ -113,7 +113,10 @@ public class ModernConfigManager {
 	public static boolean bulletBreakGlass = true;
 
 	@ConfigSync(category = CATEGORY_GAMEPLAY, comment = "A list of blocks that should be penetrable by bullets ")
-	public static String penetrableBlocks = "minecraft:glass, minecraft:glass_pane, minecraft:stained_glass, minecraft:stained_glass_pane";
+	public static String penetrableBlocks = "minecraft:glass_pane, minecraft:stained_glass_pane";
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/config/ModernConfigManager.java:116]: Please do not change these values

File: src/main/java/com/paneedah/weaponlib/config/ModernConfigManager.java [hunk 3]
```diff
@@ -113,7 +113,10 @@ public class ModernConfigManager {
...
+
+	@ConfigSync(category = CATEGORY_GAMEPLAY, comment = "Whether entities should have knockback applied to them when shot?")
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/config/ModernConfigManager.java:118]: Replace `?` by `.` just because it looks goofy (sorry)

File: Changelog.md [hunk 4]
```diff
@@ -20,6 +20,9 @@ and this project follows to [Ragnarök Versioning Convention](https://gist.githu
   - penetratesGlass
   - explosionBreak
   - penetrableBlocks
+  - knockbackOnHit
+- Make the turret work again
+- Center the firing position of the bullet when aiming
```
Comment (Desoroxxx) [inline Changelog.md:25]: Please remove those two

File: src/main/java/com/paneedah/weaponlib/EntityProjectile.java [hunk 5]
```diff
@@ -81,9 +81,9 @@ public EntityProjectile(World world, EntityLivingBase thrower, float velocity, f
     public void setPositionAndDirection(boolean isAim) {
         this.setLocationAndAngles(thrower.posX, thrower.posY + (double) thrower.getEyeHeight(), thrower.posZ, thrower.rotationYaw, thrower.rotationPitch);
         
-        this.posX -= (double) (MathHelper.cos(this.rotationYaw / 180.0F * (float) Math.PI) * 0.16F);
-        this.posY -= 0.10000000149011612D;
-        this.posZ -= (double) (MathHelper.sin(this.rotationYaw / 180.0F * (float) Math.PI) * 0.16F);
+        this.posX -= (double) (MathHelper.cos(this.rotationYaw / 180.0F * (float) Math.PI) * (isAim ? 0F : 0.16F));
+        this.posY -= (isAim ? 0D : 0.10000000149011612D);
+        this.posZ -= (double) (MathHelper.sin(this.rotationYaw / 180.0F * (float) Math.PI) * (isAim ? 0F : 0.16F));
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/EntityProjectile.java:86]: The fix to this bug should be in a different PR, it is such an important topic I want it to be easily trackable.

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

**Comment by Koud-Wind** [comment]
Please hurry up

File: src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java [hunk 6]
```diff
@@ -138,11 +139,14 @@ protected void onImpact(RayTraceResult position) {
...
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java:148]: Still please fix the spacing

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java (hunk 6 above)
Comment (Koud-Wind) [reply src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java:148]: Which line

Comment (Desoroxxx) [reply src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java:148]: 142, 147, 148.
Are you not seeing it?
image

Comment (Koud-Wind) [reply src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java:148]: image

Comment (Koud-Wind) [reply src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java:148]: There's no problem with this

Comment (Desoroxxx) [reply src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java:148]: wat is that...
Well let's just hope it's just a git bug and everything will be okay.

**Review by Desoroxxx — APPROVED** [review]
